lib/ansiblelint/errors.py

```diff
@@ -27,3 +34,19 @@ def __repr__(self):
 
         return formatstr.format(_id, self.message,
                                 self.filename, self.linenumber, self.line)
+
+    @property
+    def _hash_key(self):
+        return self.filename, self.linenumber, self.message, getattr(self.rule, 'id', 0)
+
+    def __lt__(self, other):
+        """Enable sorting of MatchError instances."""
+        return self._hash_key < other._hash_key
+
+    def __hash__(self):
+        """Perform hash of matches."""
```

Comment:
This is not a comment but a docstring: it shouldn't repeat the same thing that is written just one line below.
```suggestion
        """Return a hash value of the MatchError instance."""
```